src/Data/Simulator.elm

```diff
@@ -251,7 +277,13 @@ computeMaterialStepWaste ({ inputs, lifeCycle } as simulator) =
     in
     simulator
         |> updateLifeCycleStep Step.MaterialAndSpinning
-            (\step -> { step | mass = mass, waste = waste })
+            (\step ->
+                { step
+                    | inputMass = mass
+                    , outputMass = Quantity.difference mass waste
+                    , waste = waste
+                }
```

Comment:
I wonder if we couldn't have some `Step.updateWaste` fn which would be in charge of taking a waste, an input mass and perform the computations and assignments at the Step level? Eg

```elm
simulator
  |> updateLifeCycleStep Step.WeavingKnitting (Step.updateWaste waste mass)
```

Reply:
Addressed in https://github.com/MTES-MCT/wikicarbone/pull/27/commits/acec9efac9dedcfe9932f965ed8f234c40a77532